unittests/api_tests.cpp

```diff
@@ -2372,6 +2372,36 @@ BOOST_FIXTURE_TEST_CASE(action_ordinal_test, TESTER) { try {
    set_code( N(erin), contracts::test_api_wasm() );
    produce_blocks(1);
 
+   // prove act digest
+   auto pad = [](const digest_type& expected_act_digest, const action& act, const vector<char>& act_output)
+   {
+      std::vector<char> buf1;
+      buf1.resize(64);
+      datastream<char*> ds(buf1.data(), buf1.size());
+
+      {
+         std::vector<char> buf2;
+         const action_base* act_base = &act;
+         buf2.resize(fc::raw::pack_size(*act_base));
+         datastream<char*> ds2(buf2.data(), buf2.size());
+         fc::raw::pack(ds2, *act_base);
+         fc::raw::pack(ds, sha256::hash(buf2.data(), buf2.size()));
+      }
+
+      {
+         std::vector<char> buf2;
+         buf2.resize(fc::raw::pack_size(act.data) + fc::raw::pack_size(act_output));
+         datastream<char*> ds2(buf2.data(), buf2.size());
+         fc::raw::pack(ds2, act.data);
+         fc::raw::pack(ds2, act_output);
+         fc::raw::pack(ds, sha256::hash(buf2.data(), buf2.size()));
+      }
+
+      digest_type computed_act_digest = sha256::hash(buf1.data(), ds.tellp());
+
+      return expected_act_digest == computed_act_digest;
+   };
+
```

Comment:
How does duplicating logic of `generate_action_digest` test anything? Why not just call `generate_action_digest` ?

Reply:
`generate_action_digest` is technically a different implementation (more general) and I'm thinking we may have plans to further generalize it in the future. This more specific implementation acts as a safeguard to check for regressions/bugs as the code is further refactored.